subprojects/internal-testing/src/main/groovy/org/gradle/testing/internal/util/RetryRule.groovy

```diff
@@ -102,6 +102,7 @@ class RetryRule implements MethodRule {
     private void reRunSetup(SpecInfo spec) {
         if (spec != null) {
             reRunSetup(spec.getSuperSpec())
+            spec.getInitializerMethod()?.invoke(specification)
```

Comment:
❌ This failed because we are rerunning initializers after setup. It has to be the other way round. Sorry for missing that.